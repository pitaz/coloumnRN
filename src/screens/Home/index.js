import React from 'react';
import {View} from 'react-native';
import PropTypes from 'prop-types';
import Button from '../../components/Button';
import {styles} from './styles';

const Home = ({navigation}) => {
  const navigate = () => navigation.navigate('contacts');
  return (
    <View style={styles.homeWrapper}>
      <Button testID="contactBtn" label="Open contacts" onPress={navigate} />
    </View>
  );
};

Home.propTypes = {
  navigation: PropTypes.object,
};
export default Home;
