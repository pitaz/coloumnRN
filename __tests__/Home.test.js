import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react-native';
import Home from '../src/screens/Home';

afterEach(cleanup);

describe('Home', () => {
  let rendered;
  const navigation = {navigate: jest.fn()};
  beforeEach(() => {
    rendered = render(<Home navigation={navigation} />);
  });
  it('should match snapshot', () => {
    rendered.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders the correct message', () => {
    expect(rendered.queryByText('Open contacts')).not.toBeNull();
  });

  it('calls the open Contacts', () => {
    const buttonComponent = rendered.queryByText('Open contacts');
    fireEvent.press(buttonComponent);
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
