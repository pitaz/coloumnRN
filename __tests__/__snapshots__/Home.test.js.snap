// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#242038",
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Open contacts
    </Text>
  </View>
</View>
`;
